runtime/src/consensus/registry.rs

```diff
@@ -4,27 +4,243 @@
+    #[serde(default)]
+    pub max_apply_write_log_entries: u64,
+    #[serde(default)]
+    pub max_apply_ops: u64,
+    #[serde(default)]
+    pub checkpoint_interval: u64,
+    #[serde(default)]
+    pub checkpoint_num_kept: u64,
+    #[serde(default)]
+    pub checkpoint_chunk_size: u64,
+    #[serde(default)]
+    pub min_pool_size: u64,
+}
+
+#[derive(Clone, Debug, Default, PartialEq, Eq, Hash, Serialize, Deserialize)]
+pub struct RuntimeStakingParameters {
+    #[serde(skip_serializing_if = "Option::is_none")]
+    #[serde(default)]
+    pub thresholds: Option<BTreeMap<staking::ThresholdKind, quantity::Quantity>>,
+}
+
+#[derive(Clone, Debug, PartialEq, Eq, Hash, PartialOrd, Ord, Serialize_repr, Deserialize_repr)]
+#[repr(u32)]
+pub enum RolesMask {
+    #[serde(rename = "compute")]
+    RoleComputeWorker = 1 << 0,
+    #[serde(rename = "storage")]
+    RoleStorageWorker = 1 << 1,
+    #[serde(rename = "key-manager")]
+    RoleKeyManager = 1 << 2,
+    #[serde(rename = "validator")]
+    RoleValidator = 1 << 3,
+    #[serde(rename = "consensus-rpc")]
+    RoleConsensusRPC = 1 << 4,
+}
+
+#[derive(Clone, Debug, Default, PartialEq, Eq, Hash, Serialize, Deserialize)]
+pub struct EntityWhitelistRuntimeAdmissionPolicy {
+    #[serde(skip_serializing_if = "Option::is_none")]
+    #[serde(rename = "entities")]
+    #[serde(default)]
+    pub entities: Option<BTreeMap<PublicKey, EntityWhitelistConfig>>,
+}
+
+#[derive(Clone, Debug, Default, PartialEq, Eq, Hash, Serialize, Deserialize)]
+pub struct EntityWhitelistConfig {
+    #[serde(skip_serializing_if = "Option::is_none")]
+    #[serde(rename = "max_nodes")]
+    #[serde(default)]
+    pub max_nodes: Option<BTreeMap<RolesMask, u16>>,
+}
+
+#[derive(Clone, Debug, PartialEq, Eq, Hash, Serialize, Deserialize)]
+pub enum RuntimeAdmissionPolicy {
+    #[serde(rename = "any_node")]
+    AnyNode {},
+    #[serde(rename = "entity_whitelist")]
+    EntityWhitelist {
+        #[serde(flatten)]
+        policy: EntityWhitelistRuntimeAdmissionPolicy,
+    },
+}
+
+impl Default for RuntimeAdmissionPolicy {
+    fn default() -> Self {
+        RuntimeAdmissionPolicy::AnyNode {}
+    }
+}
+
+#[derive(Clone, Debug, PartialEq, Eq, Hash, Serialize_repr, Deserialize_repr)]
+#[repr(u8)]
+pub enum RuntimeGovernanceModel {
+    #[serde(rename = "invalid")]
+    GovernanceInvalid = 0,
+    #[serde(rename = "entity")]
+    GovernanceEntity = 1,
+    #[serde(rename = "runtime")]
+    GovernanceRuntime = 2,
+    #[serde(rename = "consensus")]
+    GovernanceConsensus = 3,
+}
+
+impl Default for RuntimeGovernanceModel {
+    fn default() -> Self {
+        RuntimeGovernanceModel::GovernanceInvalid
+    }
+}
+
+#[derive(Clone, Debug, Default, PartialEq, Eq, Hash, Serialize, Deserialize)]
+pub struct VersionInfo {
+    #[serde(default)]
+    pub version: Version,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    #[serde(with = "serde_bytes")]
+    #[serde(default)]
+    pub tee: Option<Vec<u8>>,
```

Comment:
I think you could avoid the `Option` wrapper here and instead do:
```
#[serde(skip_serializing_if = "Vec::is_empty")]
pub tee: Vec<u8>
```
and that should match the `go` serialization.

Reply:
I think the problem is in deserialization as it can be NULL.

Reply:
Yeah, sadly this doesn't work :/

Reply:
ah yeah, i was only thinking about serialization